### PR TITLE
refactor(mme): Move mme and enb stats handling to service303 thread

### LINKED ISSUE #8277: [agw] mme with high memory and service keep crashing
### Your Environment

- **Version:** 1.5.2
- **Affected Component:** AGW
- **Affected Subcomponent:** mme service
- **Deployment Environment:**bare-metal (AGW)

### Describe the Issue

mme service continuously reaches high memory and have multiple crashes in production site

```
 magma@mme.service - Magma OAI MME service
Loaded: loaded (/etc/systemd/system/magma@mme.service; disabled; vendor preset: enabled)
Active: active (running) since Fri 2021-07-23 02:28:13 UTC; 3h 2min ago
Process: 2735737 ExecStartPre=/usr/bin/env python3 /usr/local/bin/generate_oai_config.py (code=exited, status=0/SUCCESS)
Process: 2735822 ExecStartPre=/usr/bin/env python3 /usr/local/bin/config_stateless_agw.py reset_sctpd_for_stateful (code=exited, status=0/SUCCESS)
Main PID: 2735850 (mme)
Tasks: 4869 (limit: 9448)
Memory: 511.7M (limit: 512.0M)
CGroup: /system.slice/system-magma.
```

```
Jul 22 14:43:58 mme[2295491]: LeakSanitizer can not provide additional info.
Jul 22 14:43:58 mme[2295491]: SUMMARY: LeakSanitizer: SEGV /home/vagrant/magma/lte/gateway/c/oai/tasks/mme_app/mme_app_state_manager.cpp:89 in magma::lte::MmeNasStateManager::get_state(bool)
Jul 22 14:43:58 mme[2295491]: ==2295491==ABORTING
Jul 22 14:43:58 systemd[1]: magma@mme.service: Main process exited, code=exited, status=23/n/a

Jul 23 02:28:01 mme[2417820]: ==2417820==LeakSanitizer: Thread limit (8192 threads) exceeded. Dying.
Jul 23 02:28:02 systemd[1]: magma@mme.service: Main process exited, code=exited, status=23/n/a
```

**Expected behavior**
mme service shouldn't reach 512 MB memory cap and crash

### PR DESCRIPTION
**Pull request by ardzoht**
## Summary

- This contains multiple commits that are merged to master but need to be backported to v1.5 branch
- List of revisions are:
  - #7710 
  - #7758 
  - #7847
  -  #7850
  - #7853 
  - #7952 

## Test Plan

- Ran `make integ_test` on magma VM running on `v1.5` branch
- validating on debug logs statistics are being displayed:
```
============================ STATISTICS ============================================

056639 Tue Jul 27 21:25:45 2021 7F2C3D4F6700 DEBUG SERVIC tasks/service303/service303_mme_:0044                   |   Current Status|
056640 Tue Jul 27 21:25:45 2021 7F2C3D4F6700 DEBUG SERVIC tasks/service303/service303_mme_:0045    Attached UEs   |          0      |
056641 Tue Jul 27 21:25:45 2021 7F2C3D4F6700 DEBUG SERVIC tasks/service303/service303_mme_:0048    Connected UEs  |          0      |
056642 Tue Jul 27 21:25:45 2021 7F2C3D4F6700 DEBUG SERVIC tasks/service303/service303_mme_:0051    Connected eNBs |          1      |
056643 Tue Jul 27 21:25:45 2021 7F2C3D4F6700 DEBUG SERVIC tasks/service303/service303_mme_:0054    Default Bearers|          0      |
056644 Tue Jul 27 21:25:45 2021 7F2C3D4F6700 DEBUG SERVIC tasks/service303/service303_mme_:0057    S1-U Bearers   |          0      |

056645 Tue Jul 27 21:25:45 2021 7F2C3D4F6700 DEBUG SERVIC tasks/service303/service303_mme_:0061    ======================================= STATISTICS ============================================
```

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/oai/tasks/mme_app/mme_app_main.c [hunk 1]
```diff
@@ -483,6 +481,7 @@ static void* mme_app_thread(__attribute__((unused)) void* args) {
 
   // Service started, but not healthy yet
   send_app_health_to_service303(&mme_app_task_zmq_ctx, TASK_MME_APP, false);
+  start_stats_timer();
```
Comment (amarpad) [inline lte/gateway/c/oai/tasks/mme_app/mme_app_main.c:484]: This should not started in the main thread right? Can you check in master?

Comment (ardzoht) [reply lte/gateway/c/oai/tasks/mme_app/mme_app_main.c:484]: It is started on the main thread: https://github.com/magma/magma/blob/master/lte/gateway/c/core/oai/tasks/mme_app/mme_app_main.c#L522

Comment (ulaskozat) [reply lte/gateway/c/oai/tasks/mme_app/mme_app_main.c:484]: This is not the main thread.

**Review by ulaskozat — APPROVED** [review]
LGTM

**Review by themarwhal — APPROVED** [review]

**Review by pshelar — APPROVED** [review]
LGTM

**Pull request merged**